Remove legacy from tests

jbuening opened the pull request:
Get rid of some legacy code:
* Since LLVM version 3.6.0 or lit version 0.5.0, `lit_config` is the name of the global object, not `lit`.
* In `lit.cfg`, test whether the current LLVM version is in the set of known versions.
* re-enable `SolverTimeout.c` test that "occasionally fails when using Z3 4.4.1" (from 2015)

---

Other than this, I stumbled upon https://github.com/klee/klee/blob/0fd707b62988ed25c31242eb3d858895d96eb619/test/Expr/Evaluate2.kquery#L1-L5
which is in its current form since d786ba65b69c57fec0804d4a529682844ac2d411 stating "(query [false] false), which should return VALID, [...] currently returns INVALID". Indeed, Kleaver still returns INVALID, but I fail to see the reason why it should return VALID.

ccadar commented:
@jbuening this LGTM, thanks!  I only have a query about the commit adding 7.1 as known version.  Why this particular minor version?  All other versions listed there are major versions.

`(query [false] false)` should indeed output `VALID`.  That's because `false => true` in mathematical logic.  The issue here is non-trivial to address -- KLEE/Kleaver operate under the assumption that the PC is always satisfiable.  Having an option that would allow Kleaver not make this assumption would enable it to be used for arbitrary queries in the theory it supports.  As a practical consequence, we could enter it in the SMT-COMP, and see how it performs compared to its underlying solver(s).  (We did this before, but many queries would fail because of this.)

jbuening commented:
@ccadar turns out that despite specifying `LLVM_VERSION=7.0`, our TravisCI runs actually use [LLVM 7.1.0](http://lists.llvm.org/pipermail/llvm-announce/2019-May/000083.html), an ABI breaking update to 7.0.1. There are no such releases for any other version (yet). Because of this, CI failed before on this PR, and so I added it. But even if we decide to test 7.0 instead of 7.1 in CI, the check I added would prevent people using LLVM 7.1 to run the tests.

Thanks for your answer regarding the query. Still, I'm afraid I still don't understand what `(query [false] false)` is supposed to test (the [documentation](http://klee.github.io/docs/kquery/#query-commands) sadly did not make things clearer for me). I have so far not taken a close look at Kleaver, so please forgive my ignorance. Is `(query [x] y)`, supposed to query "Is `x => y` satisfiable?", which would be true (VALID) for `x=y=false`? If so, what does assuming that PC `false` is satisfiable have to do with the problem here? Also, is there any difference between `(query [false] false)` and `(query [] false)`? (I would naively assume that `[]` is the same as `[true]` since the path condition is empty?)

ccadar commented:
@jbuening thanks for clarifying the point about 7.1
For the Z3-related commit, should we keep a comment in there stating that the test might fail with Z3 4.4.1? (although it is quite an old version now)

For the query discussion, take a look at #97 and please continue the discussion there if needed.

jbuening commented:
@ccadar Sure, I think a comment doesn't hurt, so I readded it.

Thanks for the pointer to #97, that actually made things a lot clearer for me, which is why I took the liberty to add the link to the corresponding test case in this PR. Sorry for running off topic here.

ccadar commented:
Thanks again, @jbuening !
